shared/protocol/src/main/java/io/pravega/shared/metrics/ClientMetricUpdater.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.metrics;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.concurrent.ExecutorServiceHelpers;
+import io.pravega.common.concurrent.MultiKeyLatestItemSequentialProcessor;
+import java.util.concurrent.ScheduledExecutorService;
+
+
+public class ClientMetricUpdater implements MetricNotifier {
+    private final MultiKeyLatestItemSequentialProcessor<String, Long> successProcessor;
+    private final MultiKeyLatestItemSequentialProcessor<String, Long> failureProcessor;
+
+    private final ScheduledExecutorService clientMetricExecutor = ExecutorServiceHelpers.newScheduledThreadPool(2, "clientMetrics");
```

Comment:
Would one thread be sufficient?

Reply:
Yes, let's use just one. That way an implementation of a users code does not have to deal with multiple threads.